index.js

```diff
@@ -2,147 +2,87 @@
+      }));
     }
 
-    var addonTrees = this.addonTreesFor('styles');
-    var external = this._processedExternalTree();
-    var styles = new Funnel(this.trees.styles, {
-      srcDir: '/',
-      destDir: '/app/styles'
-    });
-
-    var allTrees = [this.trees.app].concat(this.addonTreesFor('app'));
-    var podStyles = new Funnel(mergeTrees(allTrees, {overwrite: true}), {
-      include: this._podStylePatterns(),
-      exclude: [ /^styles/ ],
-      destDir: '/app',
-      description: 'Funnel: Pod Styles'
+    return new Merge(concatenatedStyles, {
+      annotation: 'Merge (ember-component-css merge concatenated styles)'
     });
+  },
 
-    var trees = [external].concat(addonTrees, podStyles, styles);
+  _allowedStyleExtentions() {
+    return this.registry.extensionsForType('css');
+  },
 
-    var stylesAndVendor = mergeTrees(trees, {
-      description: 'TreeMerger (stylesAndVendor)'
-    });
+  _podDirectory() {
+    return this.appConfig.podModulePrefix ? this.appConfig.podModulePrefix.replace(this.appConfig.modulePrefix, '') : '';
+  },
 
-    var options = { outputPaths: this.options.outputPaths.app.css };
-    options.registry = this.registry;
-    var preprocessedStyles = preprocessCss(stylesAndVendor, '/app/styles', '/assets', options);
+  included(app) {
+    this._super.included(app);
+    this.appConfig = app.project.config(); 
+    this.allowedStyleExtentions = this._allowedStyleExtentions();
+  },
 
-    var vendorStyles    = this.concatFiles(stylesAndVendor, {
-      inputFiles: this.vendorStaticStyles.concat(['vendor/addons.css']),
-      outputFile: this.options.outputPaths.vendor.css,
-      description: 'Concat: Vendor Styles'
+  treeForAddon(tree) {
+    var podStyles = this._getPodStyleFunnel();
+    var podNames = new ExtractNames(podStyles, {
+      annotation: 'Walk (ember-component-css extract class names from style paths)'
     });
 
-    if (this.options.minifyCSS.enabled === true) {
-      options = this.options.minifyCSS.options || {};
-      options.registry = this.registry;
-      preprocessedStyles = preprocessMinifyCss(preprocessedStyles, options);
-      vendorStyles    = preprocessMinifyCss(vendorStyles, options);
-    }
-
-    var mergedTrees = mergeTrees([
-        preprocessedStyles,
-        vendorStyles
-      ], {
-        description: 'styles'
-      });
-    return this.addonPostprocessTree('css', mergedTrees);
-  };
-}
+    var treeAndNames = new Merge([tree, podNames], {
+      overwrite: true,
+      annotation: 'Merge (ember-component-css merge names with addon tree)'
+    });
 
-module.exports = {
-  name: 'ember-component-css',
-
-  // Gets the path to the pods folder
-  podDir: function() {
-    var podModulePrefix = this.app.project.config().podModulePrefix;
-    if (!podModulePrefix) { return 'app'; }
-    var podPath = podModulePrefix.substr(podModulePrefix.indexOf('/') + 1);
-    return path.join('app', podPath);
+    return this._super.treeForAddon.call(this, treeAndNames);
```

Comment:
I was following the instructions for these hook types [here](https://github.com/ember-cli/ember-cli/blob/master/ADDON_HOOKS.md#treefor-cont)

The passed in tree need to get modified to include other trees, and redefining the passed in argument calling super with arguments wasn't working for me.